aws/resolve_customer/resolve_customer/customer.py

```diff
@@ -0,0 +1,56 @@
+# Copyright (c) 2025 SUSE LLC.  All rights reserved.
+#
+# This file is part of suse-saas-tools
+#
+# suse-saas-tools is free software: you can redistribute it and/or modify
+# it under the terms of the GNU General Public License as published by
+# the Free Software Foundation, either version 3 of the License, or
+# (at your option) any later version.
+#
+# mash is distributed in the hope that it will be useful,
+# but WITHOUT ANY WARRANTY; without even the implied warranty of
+# MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+# GNU General Public License for more details.
+#
+# You should have received a copy of the GNU General Public License
+# along with mash.  If not, see <http://www.gnu.org/licenses/>
+#
+import logging
+import boto3
+
+from botocore.exceptions import ClientError
+
+logger = logging.getLogger()
+
+
+class AWSCustomer:
+    """
+    Get AWS customer ID information from a marketplace token
+    """
+    def __init__(self, token: str):
+        self.customer = {}
+        self.error = ''
+        if token:
+            try:
+                marketplace = boto3.client('meteringmarketplace')
```

Comment:
Since we are not handling credentials explicitly we need to have IAM permissions for the lambda function to access the AWS API. Please add this as an item to track in our tasks and issue page.

Reply:
I think these would be attached to the lambda function role

Reply:
I added the information to the issue tracker

@smarlowucf regarding access to API's from lambdas. I have done some tests with SQS and I needed to set the following AccessPolicy such that I could access the queue that I created in my account

```
 "Principal": "*",
 "Action": "SQS:*",
```

And I believe this is not the best way to do it. That access thing I leave to the SRE team and my knowledge about access policies and iam roles is very rusty

Reply:
@smarlowucf sorry I deleted a comment from you for which the above comment from me was a reaction to. Your comment appeared twice and I deleted the one from the main page, but that also deleted it here from this conversation. Very weird. Sorry about it